src/orchestrator-kubernetes/src/lib.rs

```diff
@@ -282,6 +290,60 @@ impl k8s_openapi::Metadata for PodMetrics {
     }
 }
 
+// Note that these types are very weird. We are `get`-ing a
+// `List` object, and lying about it having an `ObjectMeta`
+// (it deserializes as empty, but we don't need it). The custom
+// metrics API is designed this way, which is very non-standard.
+// A discussion in the `kube` channel in the `tokio` discord
+// confirmed that this layout + using `get_subresource` is the
+// best way to handle this.
+
+#[derive(Deserialize, Clone, Debug)]
+pub struct MetricIdentifier {
+    #[serde(rename = "metricName")]
+    pub name: String,
+    // We skip `selector` for now, as we don't use it
+}
+
+#[derive(Deserialize, Clone, Debug)]
+pub struct MetricValue {
+    #[serde(rename = "describedObject")]
+    pub described_object: ObjectReference,
+    #[serde(flatten)]
+    pub metric_identifier: MetricIdentifier,
+    pub timestamp: String,
+    pub value: Quantity,
+    // We skip `windowSeconds`, as we don't need it
+}
+
+#[derive(Deserialize, Clone, Debug)]
+pub struct MetricValueList {
+    pub metadata: ObjectMeta,
+    pub items: Vec<MetricValue>,
+}
```

Comment:
nope!

Reply:
I actually think https://docs.rs/k8s-metrics/0.12.0/k8s_metrics/custom_metrics/v1beta2/type.MetricValueList.html might just NOT work because the api uses a `List` as a resource, but definitely good to look into as a followup